src/js/components/user-management/userform.js

```diff
@@ -1,49 +1,133 @@
 import React from 'react';
+import {
+  Button,
+  Checkbox,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormControl,
+  ListItemText,
+  InputLabel,
+  MenuItem,
+  Select
+} from '@material-ui/core';
 
 import Form from '../common/forms/form';
 import TextInput from '../common/forms/textinput';
 import PasswordInput from '../common/forms/passwordinput';
-import FormButton from '../common/forms/formbutton';
+import { colors } from '../../themes/mender-theme';
 
 export default class UserForm extends React.Component {
-  componentDidUpdate(prevProps) {
-    if (this.props.editPass !== prevProps.editPass) {
+  constructor(props, context) {
+    super(props, context);
+    const isCreation = !Object.keys(props.user).length;
+    this.state = {
+      editPass: isCreation,
+      isCreation,
+      selectedRoles: props.user.roles ? props.user.roles.map(role => props.roles.find(currentRole => currentRole.id === role)) : []
+    };
+  }
+
+  componentDidUpdate(_, prevState) {
+    // TODO: this is needed due to the re-registering of inputs in the form component and should be fixed at some point
+    if (prevState.editPass !== this.state.editPass || prevState.selectedRoles !== this.state.selectedRoles) {
       this.forceUpdate();
     }
   }
 
-  handleButton() {
-    this.props.togglePass();
+  handleCheckboxClick(isSelected, role) {
+    const foundIndex = this.state.selectedRoles.findIndex(currentRole => role.id === currentRole.id);
+    let newlySelectedRoles = [...this.state.selectedRoles];
+    if (isSelected) {
+      newlySelectedRoles.push(role);
+    } else if (foundIndex > -1) {
+      newlySelectedRoles.splice(foundIndex);
+    }
+    this.setState({ selectedRoles: newlySelectedRoles });
+  }
```

Comment:
what happens now is: 
* when you click the first checkbox (atm "Admin") it checks/unchecks
* when you click the second one the list scrolls up, so it does when you check the third one
* when you check them in order from to top bottom, and then un check the first (or second) all the lower ones are unchecked as well
* when you check them from the bottom, the same automatic unchecking happens upwards
* when you start checking with the middle, the automatic unchecking happens when (un)checking the middle one
is that expected behaviour?